maintenance/errors/bricks.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2023 by PACE Telematics GmbH. All rights reserved.
+ * Created at 2023/1/20 by Sascha Voth
+ */
+
+package errors
+
+import (
+	"strconv"
+
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/pace/bricks/http/jsonapi/runtime"
+)
+
+// BricksError - a bricks err is a bricks specific error which provides
+// convenience functions to be transformed into runtime.Errors (JSON errors)
+// pb generate can be used to create a set of pre defined BricksErrors based
+// on a JSON specification, see pb generate for details
+type BricksError struct {
+	// title - a short, human-readable summary of the problem that SHOULD NOT change from occurrence
+	// to occurrence of the problem, except for purposes of localization.
+	title string
+	// detail - an application-specific error code, expressed as a string value.
+	detail string
+	// status - // the HTTP status code applicable to this problem, expressed as
+	// an int value. This SHOULD be provided.
+	status int
+	// code - application specific error code
+	code string
+}
+
+func NewBricksError(opts ...BricksErrorOption) *BricksError {
+	e := &BricksError{}
+	for _, opt := range opts {
+		opt(e)
+	}
+	return e
+}
+
+func (e *BricksError) Error() string {
+	return e.code
```

Comment:
Maybe have a method `Code() string` that returns the `code`. And in `Error() string` return something including the detail, and maybe the title too, otherwise, neither the `title` nor the `detail` ar readable.

Reply:
Getter functions have been added.